### DISCUSSION TITLE
bad optional access without access my own memory

### DISCUSSION THREAD

**Discussion by dolgae03** [ANLAB-KAIST/KENSv3#113] [Project Q&A 2. Transport Layer (TCP)] [answered]
In TestConnect_BeforeAccept, the syscall_connect is work well, but after it below error is occured.
Arrivedpacket or any system call is not called, It means they never access to my memory. Why this error happend?

![image](https://user-images.githubusercontent.com/109707681/194470547-dd0e2be7-96be-4a97-995d-ec06716489dd.png)

**Comment by lagabulin** [comment, marked as answer]
I don't know the exact reason, but I found a helpful issue about it in previous semester. I'm not sure but it came from std::optional in my opinion. You will be able to find std::optional object in a minute. I'm not a TA so it might be my best. Hope it will be helpful.
https://github.com/ANLAB-KAIST/KENSv3/discussions/92#discussion-4139747

**Comment by harryoh99** [comment]
I am encountering the same problem for days. Any suggestions for the solution..?

**Reply by dolgae03** [reply]
In my case, my code access to null pointer's argument (socket -> ~~). your code will have same problem. try to find where this null pointer access is occured!
